proto/buf/alpha/registry/v1alpha1/schema.proto

```diff
@@ -0,0 +1,196 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/descriptor.proto";
+
+// Provides endpoints for downloading dynamic schemas and for using schemas
+// in validation and data transformation functions.
+service SchemaService {
+  // GetSchema allows the caller to download a schema for one or more requested
+  // types, RPC services, or RPC methods.
+  rpc GetSchema(GetSchemaRequest) returns (GetSchemaResponse);
+
+  // Reformat allows the caller to indicate a type schema which will be used to
+  // transform given data to a different format.
+  rpc Reformat(ReformatRequest) returns (ReformatResponse);
+}
+
+// Module identifies a particular module in the registry.
+message Module {
+  // A user or organization. Required.
+  string owner = 1;
+  // A repository or module name. Required.
+  string repository = 2;
+
+  // A reference to a version. Optional. If present, it may refer to either a
+  // tag or a commit.
+  //
+  // If absent, the latest version for the module identified by the above
+  // owner and repository fields is assumed.
+  string reference = 3;
+}
+
+message GetSchemaRequest {
+  // The module from which the schema is retrieved. Required.
+  Module module = 1;
+
+  // A version in the form of a commit ID (may not be a tag or other reference).
+  // If present, the response should only include a schema if the requested
+  // version resolves to something other than this commit. This can be used
+  // to detect newer versions, when the reference in id is empty/absent, or
+  // to detect changes to a tag.
+  string only_if_not_version = 2;
+
+  // One or more element names. The names may be type names (messages or enums),
+  // service names, or method names. All names must be fully-qualified. If any name
+  // is unknown, the request will fail and no schema will be returned.
+  //
+  // The resulting schema contains these elements and all of their dependencies,
+  // which is enough information for the caller to construct a dynamic message for
+  // any requested types (e.g. requested message types or request and response types
+  // for requested service and method names).
+  repeated string element_names = 5;
+
+  // If true, well-known imports will be included the returned set of files. If
+  // false or not present, these files will omitted from the response (e.g. the
+  // caller is assumed to already know the types they define).
+  //
+  // These files are a subset of the files in the "well-known types" module:
+  // https://buf.build/beta/wellknowntypes
```

Comment:
What's the rationale behind the subset? I think its only missing plugin.proto, which is probably not that useful but feels similar to returning descriptor.proto?